### PR TITLE
Exercise three - modules

### PR DESCRIPTION
**Pull request by maz4**

### CODE REVIEW

File: js/data.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  var nearbyLisitings = [];
+  for (var i = 0; i < ADS_AMOUNT; i++) {
+    var newListing = getListingsObject(i);
+    nearbyLisitings.push(newListing);
+  }
+
+  return {
+    // ADS_AMOUNT: ADS_AMOUNT,
+    MIN_MAP_WIDTH: MIN_MAP_WIDTH,
```
Comment (mibu5) [inline js/data.js]: (OPTIONAL) Don't know if data module should expose these variables. By the name they definitely belong to the map module. Map module should be self-contained really. And this module generates data.
1) you can have a separate file that contains constant variables like this that are shared.
2) duplicate it both in map and here. This Module will go away really probably in the next PR when you add a backend call.

File: js/data.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    MAX_MAP_WIDTH: MAX_MAP_WIDTH,
+    MIN_MAP_HEIGHT: MIN_MAP_HEIGHT,
+    MAX_MAP_HEIGHT: MAX_MAP_HEIGHT,
+    nearbyLisitings: nearbyLisitings,
+
+    // getListingsObject: function (index) {
```
Comment (mibu5) [inline js/data.js]: there should not be commented code.

File: js/pin.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+window.pin = (function () {
+  var pinElementTemplate = document.querySelector('#pin').content.querySelector('.map__pin');
+
+  return {
+    generatePins: function () {
+      var pinDocFrag = document.createDocumentFragment();
+      for (var i = 0; i < window.data.nearbyLisitings.length; i++) {
+        var newPinElement = pinElementTemplate.cloneNode(true);
+        var newImgElement = newPinElement.querySelector('img');
+        newPinElement.style.left = window.data.nearbyLisitings[i].location.x + 'px';
```
Comment (mibu5) [inline js/pin.js]: create a variable `var nearbyLisiting = window.data.nearbyLisitings[i];`

and use it as `nearbyLisiting.location.x`, etc

File: js/pin.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        newPinElement.style.top = window.data.nearbyLisitings[i].location.y + 'px';
+        newPinElement.setAttribute('data-index', i);
+        newImgElement.src = window.data.nearbyLisitings[i].author.avatar;
+        newImgElement.alt = window.data.nearbyLisitings[i].offer.title;
+        pinDocFrag.appendChild(newPinElement);
+      }
+      return pinDocFrag;
+    },
+
+    pinsHandler: function (event) {
```
Comment (mibu5) [inline js/pin.js]: I feel that this module needs to just generate a pin one pin, and the rest of the logic goes to the map. And we don't need to expose window.map.mapElement.

File: js/pin.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+window.pin = (function () {
+  var pinElementTemplate = document.querySelector('#pin').content.querySelector('.map__pin');
+
+  return {
+    generatePins: function () {
+      var pinDocFrag = document.createDocumentFragment();
+      for (var i = 0; i < window.data.nearbyLisitings.length; i++) {
```
Comment (mibu5) [inline js/pin.js]: can we move this loop to the map, map has the data and then just call`window.pin.generate` locally. Just one pin,

File: js/card.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+  function escPressCloseCardHandler(event) {
+    if (event.keyCode === 27) {
+      window.card.removeCard();
+      window.card.querySelector('popup__close').removeEventListener('click', closeCardHandler);
+    }
+  }
+
+  return {
+    showCard: function (button) {
```
Comment (mibu5) [inline js/card.js]: This method should just generate the card. The rest of the logic should go to the map.

File: js/map.js [hunk 7]
```diff
@@ -329,28 +73,18 @@
...
-  }
-
-  mapElement.addEventListener('mouseup', pinsHandler);
-  propertyTypeElement.addEventListener('change', typeFieldHandler);
-  checkinElement.addEventListener('change', checkinHandler);
-  checkoutElement.addEventListener('change', checkoutHandler);
-  roomsElement.addEventListener('change', guestsHandler);
-  capacityElement.addEventListener('change', guestsHandler);
+  return {
+    mapElement: mapElement,
```
Comment (mibu5) [inline js/map.js]: don't need it

File: js/pin.js (hunk 5 above)
Comment (maz4) [reply js/pin.js]: Done

File: js/pin.js (hunk 3 above)
Comment (maz4) [reply js/pin.js]: After moving the for loop to map.js I don't have a variable "i" so I pass to the function nearbyLisiting and current count of the for loop `window.pin.generatePin(pinData, i);`

File: js/pin.js (hunk 4 above)
Comment (maz4) [reply js/pin.js]: Done

File: js/pin.js (hunk 3 above)
Comment (mibu5) [reply js/pin.js]: Why do you need to pass `i`? You need to pass the needed data to the generate, not the `i`.

File: js/pin.js (hunk 5 above)
Comment (mibu5) [reply js/pin.js]: When you say done Push the code up. I don't see any changes in the code.

File: js/card.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      var newCardElement = cardElementTemplate.cloneNode(true);
+      var imgElement = newCardElement.querySelector('.popup__photo').cloneNode(true);
+      var photosElementContainer = newCardElement.querySelector('.popup__photos');
+      var featuresElementContainer = newCardElement.querySelector('.popup__features');
+      featuresElementContainer.innerHTML = '';
+      photosElementContainer.innerHTML = '';
+
+      newCardElement.querySelector('.popup__title').textContent = offer.title;
+      newCardElement.querySelector('.popup__text--address').textContent = offer.address;
+      newCardElement.querySelector('.popup__text--price').textContent = offer.price + ' £/night';
```
Comment (mibu5) [inline js/card.js:17]: I remember you told me that you're changing pounds to yens?

File: js/pin.js (hunk 3 above)
Comment (mibu5) [reply js/pin.js]: ok got it.

**Review by mibu5 — APPROVED** [review]
Good Job!

**Pull request merged**